src/silly_sprites/opengl/buffer_object.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "../types.hpp"
+#include <glad/gl.h>
+#include <vector>
+
+namespace sly::gl {
+    class BufferObject final {
+    private:
+        using Points = std::vector<float>;
+        using Indices = std::vector<unsigned int>; // no output when in use usize so i stick to unsigned int
+
+        GLuint m_vao, m_vbo, m_ebo;
+
+        Points m_points;
+        Indices m_indices;
+
+        void init();
+        void generate_vertex_array();
+        void generate_vertex_buffer();
+        void generate_element_buffer();
+        void bind_vertex_array() const;
+        void bind_vertex_buffer() const;
+        void bind_element_buffer() const;
+        void data_vertex_buffer() const;
+        void data_element_buffer() const;
+
+    public:
+        BufferObject(Points points, Indices indices);
+        ~BufferObject();
```

Comment:
This violates the [rule of five](https://en.cppreference.com/w/cpp/language/rule_of_three#Rule_of_five).